lib/api/src/modules/url.ts

```diff
@@ -157,10 +153,43 @@ export const init: ModuleFn = ({ store, navigate, state, provider, fullAPI, ...r
   };
 
   const initModule = () => {
+    const initialArgs: Record<string, Story['args']> = {};
+
+    // Sets the args param to include any args that don't have their initial value
+    const updateArgsParam = ({ storyId, args }: { storyId: string; args?: Story['args'] }) => {
+      const customizedArgs = Object.entries(args || {}).reduce((acc, [key, value]) => {
+        if (!deepEqual(value, initialArgs[storyId][key])) acc[key] = value;
+        return acc;
+      }, {} as Story['args']);
+      const argsString = stringifyArgs(customizedArgs);
+      const argsParam = argsString.length ? `&args=${argsString}` : '';
+      queryNavigate(`${fullAPI.getUrlState().path}${argsParam}`, { replace: true });
+      api.setQueryParams({ args: argsString });
+    };
+
     fullAPI.on(NAVIGATE_URL, (url: string, options: { [k: string]: any }) => {
       fullAPI.navigateUrl(url, options);
     });
 
+    // Track initial args for the currently selected story
+    fullAPI.on(SET_STORIES, () => {
+      const data = fullAPI.getCurrentStoryData();
+      const storyArgs = isStory(data) ? data.args : undefined;
+      initialArgs[data.id] = storyArgs;
```

Comment:
Doesn't `SET_CURRENT_STORY` get called for the first story too?

Reply:
No it doesn't. It only triggers when switching between stories.